### PR TITLE
#165528858 User should receive notification when added as an admin

### PR DESCRIPTION
**Pull request by daniellamarr**
#### What does this PR do?
Currently, there's an existing implementation for changing a user's role. The aim of this PR is to check when a user's role is being changed to admin and send an email notification to that user.
#### Description of Task to be completed?
- Send email notification to newly added admin
- Ensure existing tests pass.
#### How should this be manually tested?
- Pull this branch ft-admin-receive-notification-when-added-165528858
- Make sure your application containers are running
- Run the following query
```
mutation {
  changeUserRole(email: "email@email.com", roleId: 2) {
    user {
      name
      roles{
        role
      }
    }
  }
}
```
- For the `email` field in the mutation, make use of an existing valid email address in your User model.
- For the `roleId` field in the mutation, let the id being passed tally with the id of the role admin in your Role model.
#### What are the relevant pivotal tracker stories?
[165528858](https://www.pivotaltracker.com/story/show/165528858)
#### Screenshots (if appropriate)
Screen Shot 2019-05-06 at 4 22 11 PM
Screen Shot 2019-05-06 at 4 30 31 PM

### CODE REVIEW

File: fixtures/user/user_fixture.py [hunk 1]
```diff
@@ -123,18 +123,7 @@
...
-            "user": {
-                "name": "Peter Walugembe",
-                "roles": [{
-                    "role": "Admin"
-                }]
-            }
-        }
-    }
-}
+change_user_role_mutation_response = "Role changed but email not sent"
```
Comment (TeamoreA) [inline fixtures/user/user_fixture.py]: Good work @daniellamarr . This line is repeated in `fixtures/user_role/user_role_fixtures.py`  line `113`

Comment (daniellamarr) [reply fixtures/user/user_fixture.py]: Yes, the test was written for an existing feature and for some reasons a change in my feature affected the implementation thereby making the tests fail, so I had to make the changes.

**Comment by jesseinit** [comment]
@daniellamarr the implementation works but I noticed that even when the user has been set to an admin and the email notification received, running the mutation again still send the mail notification. Is the behavior intentional?

**Comment by daniellamarr** [comment]
> @daniellamarr the implementation works but I noticed that even when the user has been set to an admin and the email notification received, running the mutation again still send the mail notification. Is the behavior intentional?

Well, I actually saw that and I had talked with Justin concerning it. The issue was from the original implementation of `change user role`, there was no check made for when a user is already assigned a role.
He said it's a bug and it should be raised on a new story because it is out of the scope of my task.

**Comment by Makwe-O** [comment]
> @daniellamarr the implementation works but I noticed that even when the user has been set to an admin and the email notification received, running the mutation again still send the mail notification. Is the behavior intentional?

Wow. That's true didn't spot that. I feel there should be a check to tell if the user is already an admin. With a proper response message

File: helpers/email/email.py [hunk 2]
```diff
@@ -63,5 +63,23 @@ def event_cancellation_notification(
...
         )
 
+    def admin_invite(self, user_email, user_name):
+        """
+        send email notification when an admin is added
+            :params
+                - user_email: the email of the user being added as an admin
+                - user_name: the name of the user being added as an admin
+        """
+        subject = 'Admin Invite'
```
Comment (ThaDeveloper) [inline helpers/email/email.py]: @daniellamarr Good work, kindly make the email subject descriptive.

Comment (TeamoreA) [reply helpers/email/email.py]: I also suppose that we are `adding` a user to be an admin rather than `inviting` them. @daniellamarr

Comment (daniellamarr) [reply helpers/email/email.py]: Thank you for the feedback, could you be of help on a more descriptive email subject?

Comment (ThaDeveloper) [reply helpers/email/email.py]: How about something in the lines of `Converge - You're now an admin` , `Role update.....`?. Something that briefly tells what's really happening.

Comment (daniellamarr) [reply helpers/email/email.py]: Alright, thank you👍

File: helpers/email/email.py [hunk 3]
```diff
@@ -63,5 +63,23 @@ def event_cancellation_notification(
             event_reject_reason=event_reject_reason
         )
 
+    def admin_invite(self, user_email, user_name):
```
Comment (Bonifase) [inline helpers/email/email.py]: Consider naming this function as `send_admin_invite_email()` if it's purpose is to send an invitation email to admins or rather use function/method names that clearly indicate what the function does and are a verb in nature like `do_something()`

File: tests/test_user/test_change_user_role.py [hunk 4]
```diff
@@ -18,7 +18,7 @@ def test_change_user_role(self):
         """
         Testing change of user role
         """
-        CommonTestCases.admin_token_assert_equal(
+        CommonTestCases.admin_token_assert_in(
```
Comment (Bonifase) [inline tests/test_user/test_change_user_role.py]: Consider adding test for changing already existing user role.

File: helpers/email/email.py (hunk 3 above)
Comment (daniellamarr) [reply helpers/email/email.py]: Alright, I will change that.

File: helpers/email/email.py (hunk 2 above)
Comment (daniellamarr) [reply helpers/email/email.py]: Done

File: helpers/email/email.py (hunk 3 above)
Comment (daniellamarr) [reply helpers/email/email.py]: Done

File: tests/test_user/test_change_user_role.py (hunk 4 above)
Comment (daniellamarr) [reply tests/test_user/test_change_user_role.py]: Done

File: tests/base.py [hunk 5]
```diff
@@ -313,3 +313,15 @@ def func_wrapper(self):
         db_session().commit()
         return headers
     return func_wrapper
+
+
+def change_role_test_user(func):
```
Comment (Bonifase) [inline tests/base.py]: Rename this to `change_test_user_role()`

File: tests/test_user/test_change_user_role.py [hunk 6]
```diff
@@ -50,3 +53,13 @@ def test_assign_role_to_none_existing_user(self):
             assign_role_to_non_existing_user_mutation,
             "User not found"
         )
+
+    def test_change_user_role_with_already_assigned_role(self):
+        """
+        Testing change of user role
```
Comment (Bonifase) [inline tests/test_user/test_change_user_role.py]: This description is vague. It does not tell what aspect is being tested when changing user role.

**Pull request merged**